### PR TITLE
Cashout Improvements

### PR DESCRIPTION
**Pull request by Hugo0**
# Bank Account Input and Validation Improvements

## Core Changes
- Standardized bank account input handling across all components
- Improved US bank account validation and formatting
- Added proper handling of existing account addresses for KYC'd users
- Reduced minimum cashout amount to $1 in development environment
- added test suite and pnpm commands

### CODE REVIEW

File: src/components/Cashout/Components/Initial.view.tsx [hunk 1]
```diff
@@ -220,11 +221,35 @@ export const InitialCashoutView = ({
...
+
+        if (storedAccount.account_type === 'iban') {
+            return sanitizedInput === sanitizedStored
+        } else if (storedAccount.account_type === 'us') {
+            // For US accounts, only match against the account number part (after routing number)
+            const storedAccountNumber = sanitizedStored.slice(9)
+            return sanitizedInput === storedAccountNumber
+        }
+        return false
+    }
```
[the remark this answers is not in the document]
Comment (Hugo0) [reply src/components/Cashout/Components/Initial.view.tsx:251]: wrong

File: src/components/Global/GeneralRecipientInput/index.tsx [hunk 2]
```diff
@@ -40,17 +41,15 @@ const GeneralRecipientInput = ({
...
                 if (!isValid) errorMessage.current = 'Invalid IBAN, country not supported'
-            } else if (/^[0-9]{6,26}$/.test(recipient)) {
+            } else if (/^[0-9]{1,17}$/.test(sanitizedInput)) {
                 type = 'us'
-                isValid = await utils.validateBankAccount(recipient)
-                if (!isValid) {
-                    errorMessage.current =
-                        'Invalid bank account. For US bank accounts, enter your bank routing number (9 digits) followed by your account number (example: 1112223330001234567 where routing number is: 111222333 and account number: 0001234567)'
-                }
+                isValid = true
```
[the remark this answers is not in the document]
Comment (Hugo0) [reply src/components/Global/GeneralRecipientInput/index.tsx:52]: provide it

File: src/components/Global/GeneralRecipientInput/index.tsx [hunk 3]
```diff
@@ -105,19 +104,28 @@
...
-            debounceTime={750}
+            placeholder={placeholder}
             validate={checkAddress}
             onUpdate={onInputUpdate}
             className={className}
             autoComplete="on"
             name="bank-account"
             suggestions={getSuggestions(recipientType.current)}
             infoText={infoText}
+            formatDisplayValue={formatDisplayValue}
```
[the remark this answers is not in the document]
Comment (Hugo0) [reply src/components/Global/GeneralRecipientInput/index.tsx:131]: Input needs to be reworked for: autocomplete="bank-account-number"

File: src/components/Offramp/Success.view.tsx [hunk 4]
```diff
@@ -57,40 +57,40 @@ export const OfframpSuccessView = ({
...
-                                        offrampForm.recipient.replaceAll(/\s/g, '').toLowerCase()
-                                )?.account_type === 'iban'
+                        <div className="flex items-center gap-1">
+                            $
+                            {user?.accounts.find(
+                                (account) =>
+                                    account.account_identifier.replaceAll(/\s/g, '').toLowerCase() ===
+                                    offrampForm.recipient.replaceAll(/\s/g, '').toLowerCase()
+                            )?.account_type === 'iban'
+                                ? offrampType == _consts.OfframpType.CASHOUT
```
Comment (Hugo0) [inline src/components/Offramp/Success.view.tsx:72]: this needs to be standardized with new format utils

File: .env.example [hunk 5]
```diff
@@ -1,20 +1,23 @@
...
-export NEXT_PUBLIC_BASE_URL="https://peanut.to"
+export PEANUT_API_KEY="UtyraajNKZqCelV3k2U0Y1xW7l3WwWpI" # DEMO KEY
+
+export NEXT_PUBLIC_PEANUT_ACCESS_CODE="ilovepeanuts"
+export NEXT_PUBLIC_NOTIFY_API_SECRET="1ad9fc39-f513-4382-be29-e591d1ac5fd7"
+export NEXT_PUBLIC_GA_KEY="G-KK0SP9TQCF"
+export NEXT_PUBLIC_WC_PROJECT_ID="6b12debabfec7d3971d2befb5a2e36fd"
+export NEXT_PUBLIC_SENTRY_DSN="https://d2f7d5e16a3c2cecdd301c7f94e440d0@o4505827429187584.ingest.sentry.io/4505827431415808"
+export NEXT_PUBLIC_RECAPTCHA_SITE_KEY="6Ldio2kpAAAAAAWJiNHbgv1yCzsROdxU3_Py85rZ"
+export NEXT_PUBLIC_INFURA_API_KEY="051209b4dbce4a98bfe8b424c6b21417"
```
[the remark this answers is not in the document]
Comment (jjramirezn) [reply .env.example]: @Hugo0 any reason for publishing these keys

**Comment by Hugo0** [comment]
I think there's still a bunch of stuff to improve here. Don't have time to tackle all of these, but:
- we should reuse confirm / success screens between link offramp and self-cashout. we need DRY here
- Have a lot of separate consts in different files. Need to reuse & consolidate here too
- too much state in link claim view
- need to split up claiming components into 4 big buttons that lead to different flows  (bank, address/ens, CEX etc)

File: prettier.config.js [hunk 6]
```diff
@@ -5,12 +5,4 @@ module.exports = {
     semi: false,
     singleQuote: true,
     printWidth: 120,
-    importOrder: [
```
Comment (jjramirezn) [inline prettier.config.js:8]: Yeeeeessss

File: src/components/Global/ValidatedInput/index.tsx [hunk 7]
```diff
@@ -76,8 +79,13 @@ const ValidatedInput = ({
     }, [value, debounceTime])
 
     const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
-        const newValue = e.target.value
-        onUpdate({ value: newValue, isValid: false, isChanging: !!newValue })
+        const rawValue = e.target.value
+        const sanitizedValue = sanitizeBankAccount(rawValue)
```
Comment (jjramirezn) [inline src/components/Global/ValidatedInput/index.tsx]: Why is this here, this component is used not only for bank accounts

File: .env.example (hunk 5 above)
Comment (jjramirezn) [inline .env.example]: Should we publish these?

**Review by jjramirezn — APPROVED** [review]
Changes seem good, lots of complexity around cashout

One comment about the ValidatedInput please review it before merging.
